overlord/ifacestate/ifacestate_test.go

```diff
@@ -893,10 +893,18 @@ plugs:
   interface: network
 `
 
-// The setup-profiles task will not auto-connect an plug that was previously
+// The setup-profiles task will not auto-connect a plug that was previously
 // explicitly disconnected by the user.
-func (s *interfaceManagerSuite) TestDoSetupSnapSecurityHonorsDisconnect(c *C) {
-	c.Skip("feature disabled until redesign/reimpl")
+func (s *interfaceManagerSuite) TestDoSetupSnapSecurityHonorsUndesiredAutoConnect(c *C) {
+	s.state.Lock()
+	s.state.Set("conns", map[string]interface{}{
+		"snap:network ubuntu-core:network": map[string]interface{}{
+			"auto":      true,
```

Comment:
I think the auto-connect aspect is irrelevant here. We can just drop this line and the test is as relevant as before.